query_optimizer/ExecutionGenerator.cpp

```diff
@@ -1030,9 +1184,20 @@ void ExecutionGenerator::convertCreateTable(
     catalog_relation->setDefaultStorageBlockLayout(layout.release());
   }
 
+  if (physical_plan->partition_scheme()) {
+    LOG(INFO) << "Relation created with partition scheme." << std::endl;
+    std::size_t num_partitions = physical_plan->partition_scheme()->getPartitionSchemeHeader().getNumPartitions();
+    attribute_id part_attr_id = physical_plan->partition_scheme()->getPartitionSchemeHeader().getPartitionAttributeId();
```

Comment:
Please mark above two `const`.

And save `physical_plan->partition_scheme()->getPartitionSchemeHeader()` as we use it twice here.